src/box/read_view.h

```diff
@@ -12,6 +12,8 @@
 #include "small/rlist.h"
 #include "vclock/vclock.h"
 
+#include "engine.h"
+
```

Comment:
It seems you only added this include to be able to declare a pointer to `struct engine`.

For such a purpose better declare `struct engine;` like this. Loot at how it's done a couple of lines below:
https://github.com/tarantool/tarantool/blob/fbaaa140a9817ba091e1a1874970150c958fb72f/src/box/read_view.h#L19-L22